pkg/util/aad/aad.go

```diff
@@ -57,14 +61,13 @@ func GetToken(ctx context.Context, log *logrus.Entry, oc *api.OpenShiftCluster,
 	wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
 		err = token.RefreshWithContext(ctx)
 		if err != nil {
-			isAADSTS700016 := strings.Contains(err.Error(), "AADSTS700016")
+			log.Info(err)
 
 			// populate err with a user-facing error that will be visible if
 			// we're not successful.
-			log.Info(err)
 			err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalCredentials, "properties.servicePrincipalProfile", "The provided service principal credentials are invalid.")
 
-			if isAADSTS700016 {
+			if !autorest.IsTokenRefreshError(err) || strings.Contains(err.Error(), "AADSTS700016") {
```

Comment:
this is not valid - you have overwritten err by this point.

Reply:
Argh, right! I was goign to do a separate variable `isTokenRefreshError` initially for that reason. Tests will help to catch things like this, but it is going to be tricky to them it without refactoring. Will see what I can do when I have a moment for it.